### ISSUE TITLE
Template with GDNative enabled for Javascript fails building.

### ISSUE THREAD

**Issue by feserr** [godotengine/godot#54957]
### Godot version

3.4-stable

### System information

macOS 11.6.1, Emscripten 2.0.34

### Issue description

Building the templates for Javascript with GDNative enable fails to compile with the Godot version 3.4 using Emscripten 2.0.34

The error is the following:
```bash
$ scons platform=javascript tools=no gdnative_enabled=yes target=release
scons: Reading SConscript files ...
WebM SIMD optimizations are disabled. Check if your CPU architecture, CPU bits or platform are supported!
Checking for C header file mntent.h... (cached) yes
scons: done reading SConscript files.
scons: Building targets ...
[  0%] Install file: "bin/godot.javascript.opt.gdnative.wrapped.js" as "bin/.javascript_zip/godot.js"
[ 75%] Linking Program        ==> bin/godot.side.javascript.opt.gdnative.wasm
em++: error: EM_ASM is not supported in side modules
scons: *** [bin/godot.side.javascript.opt.gdnative.wasm] Error 1
scons: building terminated because of errors.
[Time elapsed: 00:00:17.289]
```

It works fine if it is builded without the gdnative enabled.

### Steps to reproduce

- Clone the repo.
- Source the environmental variables of Emscripten.
- Run the command: `scons platform=javascript tools=no gdnative_enabled=yes target=release`

**Comment by Calinou** [comment]
cc @Faless

**Comment by akien-mga** [comment]
Fixed by #54973.
